doc/source/entrypoints.rst

```diff
@@ -0,0 +1,45 @@
+Entry points
+============
+
+The (de-)serialization methods registered with ``fsc.hdf5-io`` are only available once the Python module defining them has been loaded. To avoid having to explicitly ``import`` all necessary modules before loading a module, ``fsc.hdf5-io`` defines two *entry point groups*:
+
+Serialization: ``fsc.hdf5_io.save``
+-----------------------------------
+
+When a Python object whose serialization is not defined is encountered, ``fsc.hdf5-io`` will load (if it exists) the entry point corresponding to the full name of the object's class in the ``fsc.hdf5_io.save`` entrypoint. If there is no entry point for the _exact_ Python name, it will also try its module name(s).
+
+For example, if a ``scipy.sparse.csr.csr_matrix`` should be serialized, it will first check for an entry point named ``scipy.sparse.csr.csr_matrix`` in ``fsc.hdf5_io.save``. If this entry point does not exist, it will try ``scipy.sparse.csr``, ``scipy.sparse``, and ``scipy``, stopping at the first entry point that exists.
+
```

Comment:
The module itself is always loaded already - because we have an object of that module at our hands.

Reply:
This is why the `save` is probably much less likely to be used - it only affects "special type" implementations where you don't have the possibility to implement it in the class itself.

Reply:
Is it a large effort to first check the module and then hdf5_io?

Reply:
Not really, but I don't see the point. All this does is import the module defined in the entrypoint, and then go through the regular mechanism again.

If I have an object, for example a ``csr_matrix``, then ``scipy.sparse.csr`` is _already_ imported, otherwise I couldn't have this object. Is you suggestion to explicitly import also ``scipy`` and ``scipy.sparse``?

Reply:
Ah, I was thinking on load.
It desers an object it does not have the type_tag for and then looks up entry points, right?
If it looks first in hdf5_io and finds it, it wont look further. 

But maybe my special_sympy module is not happy with your sympy load/save, shouldnt it be found first? Because on save, it will use its own save function.

I guess it is not a problem as long as I dont use your type_tags, but that still means that in the single dispatch, we have two times the sympy.target_class, once yours, once mine...

Reply:
Well, this isn't really a solvable problem: On the one hand, we want to be able to define serialization for "special" types that don't "belong" to our module. But then also, of course _everybody_ can do the same. At the level of `fsc.hdf5-io`, there's not way of distinguishing which is the one you want.

If you really _need_ to override the saving, you need to make sure to register your own singledispatch functions _after_ the existing ones.

Reply:
Makes sense, or (safer) subclass entirey

Reply:
Basically, it boils down to: If I want anybody to be able to register serializers for any type, I need to accept that anybody can register serializers for any type...

It's a design tradeoff, where we've gone with leaving it completely open. The assumption is that different packages using `fsc.hdf5-io` will collaboratively define how to serialize special types. 

As a side note, I don't think I've implemented special types anywhere other than in the base package.

Reply:
Fair, this point can stay the way it is. I will just test the entry points on a small module and then we're fine